test/e2e/federated-service.go

```diff
@@ -369,22 +453,24 @@ func discoverService(f *framework.Framework, name string, exists bool) {
 	}
 
 	_, err := f.Client.Pods(f.Namespace.Name).Create(pod)
-	Expect(err).NotTo(HaveOccurred(), "Trying to create pod to run %q", command)
-	defer f.Client.Pods(f.Namespace.Name).Delete(FederatedServicePod, api.NewDeleteOptions(0))
+	framework.ExpectNoError(err, "Trying to create pod to run %q", command)
+	defer func() { f.Client.Pods(f.Namespace.Name).Delete(podName, api.NewDeleteOptions(0)) }()
 
 	if exists {
 		// TODO(mml): Eventually check the IP address is correct, too.
-		Eventually(podExitCodeDetector(f, FederatedServicePod, 0), 10*DNSTTL, time.Second*2).
+		Eventually(podExitCodeDetector(f, podName, 0), 2*DNSTTL, time.Second*2).
```

Comment:
OK, will do.  But given that this is using a brand new DNS name, this should be immediate.  I'll open an issue to figure out why it is not immediate.  I suspect a bug in the Service Controller.